### PR TITLE
New input API

### LINKED ISSUE #8: Input format
Three.js doesn't appear to provide a single structure for all the input per frame.
Main problem is - some use cases need an impulse mode (has "X" just been hit?), some need a modal one (is "x" currently pressed?). We night end up doing both.

### PR DESCRIPTION
**Pull request by kvark**
Closes #8

r? @vitvakatu

### CODE REVIEW

File: src/input.rs [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+    is_focused: bool,
+    keys_pressed: HashSet<Key>,
+    mouse_pressed: HashSet<MouseButton>,
+    mouse_pos: mint::Point2<f32>,
+}
+
+#[allow(dead_code)]
+struct InputDiff {
+    time_delta: TimerDuration,
+    //time_delta_focused: time::Duration,
```
Comment (vitvakatu) [inline src/input.rs]: Do we need this?

File: src/input.rs [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+        });
+        self.0.mouse_pos = pos;
+    }
+}
+
+
+#[derive(Clone, Copy, Debug, PartialEq)]
+pub struct Timer {
+    start: time::Instant,
+    //focus_only: bool,
```
Comment (vitvakatu) [inline src/input.rs]: And this?

File: src/input.rs [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+    Mouse(MouseButton),
+}
+
+pub const KEY_ESCAPE: Button = Button::Key(Key::Escape);
+pub const KEY_SPACE: Button = Button::Key(Key::Space);
+
+impl Button {
+    pub fn count_hits(&self, input: &Input) -> u8 {
+        match *self {
+            Button::Key(button) => input.1.keys_hit.iter().filter(|&&key| key == button).take(0xFF).count() as u8,
```
Comment (vitvakatu) [inline src/input.rs]: Any reason to use hex number instead of decimal?
Also, maybe we should use constant for this?

File: src/input.rs [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+
+
+#[derive(Clone, Copy, Debug, PartialEq)]
+pub struct KeyAxis {
+    pub neg: Key,
+    pub pos: Key,
+}
+
+pub const AXIS_LEFT_RIGHT: KeyAxis = KeyAxis{ neg: Key::Left, pos: Key::Right };
+pub const AXIS_DOWN_UP: KeyAxis = KeyAxis{ neg: Key::Down, pos: Key::Up };
```
Comment (vitvakatu) [inline src/input.rs]: It seems irregular for me. Maybe `AXIS_UP_DOWN`? People usually look from the top to the down.

File: src/input.rs [hunk 5]
```diff
@@ -0,0 +1,172 @@
+#![allow(missing_docs)] //TODO
```
Comment (vitvakatu) [inline src/input.rs]: I'll try to fix it asap.

Comment (kvark) [inline src/input.rs]: thanks, and no rush!

File: src/input.rs (hunk 1 above)
Comment (kvark) [inline src/input.rs]: removed now

File: src/input.rs (hunk 3 above)
Comment (kvark) [inline src/input.rs]: fixed

File: src/input.rs (hunk 4 above)
Comment (kvark) [inline src/input.rs]: The axis go from DOWN (negative) to UP (positive), hence the name `AXIS_DOWN_UP`, which is consistent with `AXIS_LEFT_RIGHT`.

**Comment by kvark** [comment]
Comments addressed, unleashing bors.
bors r+

**Pull request merged**